File: client-scalaz/src/main/scala/org/openapitools/client/api/RegionResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import RegionResponse._

case class RegionResponse (
  id: Option[Long],
shortName: Option[String],
fullName: Option[String],
parent: Option[RegionResponse],
children: Option[List[RegionResponse]],
postalCodes: Option[List[PostalCodeResponse]],
latitude: Option[Double],
longitude: Option[Double],
active: Option[Boolean],
root: Option[Boolean],
regionClass: Option[String],
start: Option[Long],
end: Option[Long],
polygon: Option[String],
metaData: Option[String],
distance: Option[Double],
versionCode: Option[Integer])

object RegionResponse {
  import DateTimeCodecs._

  implicit val RegionResponseCodecJson: CodecJson[RegionResponse] = CodecJson.derive[RegionResponse]
  implicit val RegionResponseDecoder: EntityDecoder[RegionResponse] = jsonOf[RegionResponse]
  implicit val RegionResponseEncoder: EntityEncoder[RegionResponse] = jsonEncoderOf[RegionResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/region_response.clj
(ns sirqul-io-t-platform.specs.region-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.region-response :refer :all]
            [sirqul-io-t-platform.specs.region-response :refer :all]
            [sirqul-io-t-platform.specs.postal-code-response :refer :all]
            )
  (:import (java.io File)))


(def region-response-data
  {
   (ds/opt :id) int?
   (ds/opt :shortName) string?
   (ds/opt :fullName) string?
   (ds/opt :parent) region-response-spec
   (ds/opt :children) (s/coll-of region-response-spec)
   (ds/opt :postalCodes) (s/coll-of postal-code-response-spec)
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :active) boolean?
   (ds/opt :root) boolean?
   (ds/opt :regionClass) string?
   (ds/opt :start) int?
   (ds/opt :end) int?
   (ds/opt :polygon) string?
   (ds/opt :metaData) string?
   (ds/opt :distance) float?
   (ds/opt :versionCode) int?
   })

(def region-response-spec
  (ds/spec
    {:name ::region-response
     :spec region-response-data}))
